python/ray/data/dataset_pipeline.py

```diff
@@ -562,6 +563,35 @@ def schema(
             self._schema = self._peek().schema(fetch_if_missing)
         return self._schema
 
+    def dataset_format(self) -> BlockFormat:
+        """The format of the dataset pipeline's underlying data blocks. Possible
+        values are: "arrow", "pandas" and "simple".
+
+        This may block; if the schema is unknown, this will synchronously fetch
+        the schema for the first block.
+        """
+        # We need schema to properly validate, so synchronously
+        # fetch it if necessary.
+        schema = self.schema(fetch_if_missing=True)
```

Comment:
With the pipeline peeking implemented above, this triggering execution should be fine, i.e. we shouldn't hit the double-execution issue. 👍